src/lifecycles/RoomViewLifecycle.ts

```diff
@@ -14,15 +14,29 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+import React from "react";
+
 export enum RoomViewLifecycle {
     PreviewRoomNotLoggedIn = "preview_not_logged_in",
     JoinFromRoomPreview = "try_join_not_logged_in",
+    RenderRoomHeader = "render_room_header",
```

Comment:
Could we have a more generic `view_room` lifecycle method? 

There's nothing inherently related to the room header here. And it could be reused in more context